src/framework/components/anim/binder.js

```diff
@@ -0,0 +1,249 @@
+Object.assign(pc, function () {
+
+    var AnimComponentBinder = function (animComponent, graph) {
+        this.animComponent = animComponent;
+
+        if (graph) {
+            pc.DefaultAnimBinder.call(this, graph);
+        } else {
+            this.propertyLocator = new pc.AnimPropertyLocator();
+        }
+    };
+
+    AnimComponentBinder.prototype = Object.create(pc.DefaultAnimBinder.prototype);
+    AnimComponentBinder.prototype.constructor = AnimComponentBinder;
+
+    Object.assign(AnimComponentBinder.prototype, {
+        resolve: function (path) {
+            var pathSections = this.propertyLocator.decode(path);
+
+            var entityHierarchy = pathSections[0];
+            var component = pathSections[1];
+            var propertyHierarchy = pathSections[2];
+
+            var entity = this._getEntityFromHierarchy(entityHierarchy);
+
+            if (!entity)
+                return null;
+
+            var propertyComponent;
+
+            switch (component) {
+                case 'entity':
+                    propertyComponent = entity;
+                    break;
+                case 'graph':
+                    propertyComponent = this.nodes[entityHierarchy[0]].node;
+                    break;
+                default:
+                    propertyComponent = entity.findComponent(component);
+                    if (!propertyComponent)
+                        return null;
+            }
+
+            return this._createAnimTargetForProperty(propertyComponent, propertyHierarchy);
+        },
+
+        update: function (deltaTime) {
+            // flag active nodes as dirty
+            var activeNodes = this.activeNodes;
+            if (activeNodes) {
+                for (var i = 0; i < activeNodes.length; ++i) {
+                    activeNodes[i]._dirtifyLocal();
+                }
+            }
+        },
+
+        _getEntityFromHierarchy: function (entityHierarchy) {
+            if (!this.animComponent.entity.name === entityHierarchy[0]) {
+                return null;
+            }
+
+            var currEntity = this.animComponent.entity;
+
+            if (entityHierarchy.length === 1) {
+                return currEntity;
+            }
+            return currEntity._parent.findByPath(entityHierarchy.join('/'));
```

Comment:
Just FYI - I believe `findByPath` is sloooooow, so if this is performance-sensitive code, this may become an issue. Not blocking for merging this PR tho, IMHO.

Reply:
Ok, i'd say we can test this then. The binder runs this once when the animation is first loaded and before it's played.